oci/spec_opts.go

```diff
@@ -377,6 +377,14 @@ func WithImageConfigArgs(image Image, args []string) SpecOpts {
 			return fmt.Errorf("unknown image config media type %s", ic.MediaType)
 		}
 
+		//FreeBSD has support for running Linux binaries via syscall redirection, while that would be enough
+		//for some binaries others depend on emulated linux file systems (procfs & sysfs).
+		//So when we are running FreeBSD and our image OS is Linux we modify our mounts to mount the emulated
+		//filesystems.
+		//We do it here since here we have access to both the OCI spec & the image OS.
+		if runtime.GOOS == "freebsd" && ociimage.OS == "linux" {
+			freebsdLinuxEmulationMounts(s)
+		}
```

Comment:
Carrying over the comment I left on the previous PR:

I'd like to see this extracted out into a function like `appendOSMounts`, which is then separately defined in each of `spec_opts_freebsd.go`/`spec_opts_linux.go`/`spec_opts_windows.go` with whatever content is appropriate there. The Linux one would likely be empty, but I imagine there could be some use for this on Windows with the LCOW/WCOW work.

In addition to providing a possibly-reusable abstraction for FreeBSD and Windows to achieve similar goals, an advantage is keeping FreeBSD-specific filesystems like "linprocfs" and "linsysfs" out of the common file that is compiled for all operating systems.

I'd also like to see a `WithLinuxEmulationMounts` functional option that clients of containerd can use without having to use `WithImageConfig`/`WithImageConfigArgs` or even having an image with a declared OS of "linux".

If you're not interested in doing that work, I can pick up and carry this PR.

Reply:
> I'd like to see this extracted out into a function like appendOSMounts

That makes a lot of sense, implemented. As for `WithLinuxEmulationMounts` I'm not that familiar with the codebase yet, so I would defer it to you

Reply:
That works for me, I can pick up `WithLinuxEmulationMounts` in a follow-up PR.